upload-processor/BulkFileUploadFunctionApp/Services/UploadProcessingService.cs

```
@@ -0,0 +1,652 @@
+
+            _destinationAndEvents = GetAllDestinationAndEvents();
+
+            _uploadEventHubService = uploadEventHubService;
+            _dexStorageAccountConnectionString = $"DefaultEndpointsProtocol=https;AccountName={_dexAzureStorageAccountName};AccountKey={_dexAzureStorageAccountKey};EndpointSuffix=core.windows.net";
+        }
+
+        /// <summary>
+        /// Processeses the given blob created event from the URL provided.
+        /// </summary>
+        /// <param name="blobCreatedUrl"></param>
+        /// <returns></returns>
+        /// <exception cref="Exception"></exception>
+        public async Task ProcessBlob(string blobCreatedUrl)
+        {
+            _logger.LogInformation($"TUS_AZURE_OBJECT_PREFIX={_tusAzureObjectPrefix}, TUS_AZURE_STORAGE_CONTAINER={_tusAzureStorageContainer}, DEX_AZURE_STORAGE_ACCOUNT_NAME={_dexAzureStorageAccountName}");
+
+            Trace? trace = null;
+            Span? copySpan = null;
+
+            string? destinationContainerName = null;
+            string? destinationBlobFilename = null;
+            Dictionary<string, string> tusFileMetadata = null;
+            string? destinationId = null;
+            string? eventType = null;
+
+            string? dexBlobUrl = null;
+
+            try
+            {
+                _logger.LogInformation($"Processing blob url: {blobCreatedUrl}");
+
+                var sourceBlobUri = new Uri(blobCreatedUrl);
+                string tusInfoFilename = sourceBlobUri.Segments.Last();
+
+                _logger.LogInformation($"tusPayloadFilename is = {tusInfoFilename}");
+
+                var tusPayloadPathname = $"/{_tusAzureObjectPrefix}/{tusInfoFilename}";
+                
+                TusInfoFile tusInfoFile = await GetTusFileInfo(tusInfoFilename);
+
+                if (tusInfoFile.ID == null)
+                {
+                    _logger.LogError("Malformed tus info file.  No ID provided.");
+                    return;
```

Throw exception here instead of return early.